com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/AbortFailedProcessCommandFactory.java

```diff
@@ -2,42 +2,48 @@
 
 import java.text.MessageFormat;
 
+import com.sap.cloud.lm.sl.cf.process.flowable.AbortProcessAction;
+import com.sap.cloud.lm.sl.cf.process.flowable.ProcessAction;
+import com.sap.cloud.lm.sl.cf.process.flowable.ProcessActionRegistry;
 import org.apache.commons.lang3.BooleanUtils;
 import org.flowable.common.engine.impl.interceptor.Command;
 import org.flowable.common.engine.impl.interceptor.CommandContext;
-import org.flowable.engine.RuntimeService;
-import org.flowable.engine.impl.context.Context;
 import org.flowable.engine.impl.jobexecutor.DefaultFailedJobCommandFactory;
 import org.flowable.engine.impl.util.CommandContextUtil;
 import org.flowable.job.service.impl.persistence.entity.JobEntity;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import com.sap.cloud.lm.sl.cf.process.util.HistoryUtil;
-import com.sap.cloud.lm.sl.cf.web.api.model.Operation;
+import org.springframework.context.annotation.Lazy;
 
+import javax.inject.Inject;
+import javax.inject.Named;
+
+@Named
+@Lazy
```

Comment:
Why @ Lazy ? It is initialized at the startup of the deploy-service.

Reply:
Because there are circular dependencies when the component is initialiazed eagerly